ppl/detail/Correlation.tsx

```diff
@@ -0,0 +1,54 @@
+import {useSelector} from "react-redux"
+import React, {memo, useMemo} from "react"
+
+import {reactElementProps} from "src/js/test/integration"
+import LogDetails from "src/js/state/LogDetails"
+import {zng} from "zealot"
+import {Caption, ChartWrap, TableWrap} from "app/detail/Shared"
+import PanelHeading from "app/detail/PanelHeading"
+import EventTimeline from "ppl/detail/EventTimeline"
+import {BrimEvent} from "ppl/detail/models/BrimEvent"
+import {sort} from "ppl/detail/util/sort"
+import {Data, Name, Value} from "app/core/Data"
+import formatDur from "ppl/detail/util/formatDur"
+import {isEqual} from "lodash"
+import Panel from "app/detail/Panel"
+import {getCorrelationQuery} from "./flows/getCorrelationQuery"
+import EventLimit from "./EventLimit"
+
+export default memo(function UidPanel({record}: {record: zng.Record}) {
+  const query = useMemo(() => getCorrelationQuery(record), [record])
+  const isLoading = useSelector(LogDetails.getUidStatus) === "FETCHING"
+  const logs = useSelector(LogDetails.getUidLogs)
+
+  const events = useMemo(() => {
+    return sort(logs).map(BrimEvent.build)
+  }, [logs])
+
+  const conn = useMemo(() => {
+    return events.find((e) => e.getType() === "conn")
+  }, [events])
+
+  const index = useMemo(() => {
+    return events.findIndex((e) => isEqual(e.getRecord(), record))
+  }, [record, events])
+
+  return (
+    <section {...reactElementProps("correlationPanel")}>
+      <PanelHeading isLoading={isLoading}>Correlation</PanelHeading>
+      <Panel isLoading={isLoading && events.length === 0}>
+        <ChartWrap>
+          <EventTimeline events={events} current={index} />
+          <EventLimit query={query} count={events.length} />
+        </ChartWrap>
+        <TableWrap>
+          <Data>
+            <Name>Duration</Name>
+            <Value>{formatDur(conn?.getTime(), conn?.getEndTime())}</Value>
+          </Data>
+        </TableWrap>
+      </Panel>
+      <Caption>Populated by uid & community_id</Caption>
```

Comment:
This should just be "Populated by uid", no?

Reply:
This one uses both, uid to get the Zeek records, and then if there is a conn in that first set, use it to get any associated alerts during that time:  Here's the function that gets those records: https://github.com/brimsec/brim/blob/3dadc9226574de87bfa8cef2251569061217588d/ppl/detail/flows/fetch.ts#L18